src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -0,0 +1,477 @@
+package cloud.eppo;
+
+import static cloud.eppo.Utils.getMD5Hex;
+import static cloud.eppo.Utils.throwIfEmptyOrNull;
+
+import cloud.eppo.logging.Assignment;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.FlagConfig;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.VariationType;
+import java.util.HashMap;
+import java.util.Map;
+
+import cloud.eppo.ufc.dto.adapters.EppoModule;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EppoClient {
+  private static final Logger log = LoggerFactory.getLogger(EppoClient.class);
+  private final ObjectMapper mapper = new ObjectMapper().registerModule(EppoModule.eppoModule()); // TODO: is this the best place for this?
+  private static final String DEFAULT_HOST = "https://fscdn.eppo.cloud";
```

Comment:
For now, the tests are using our [cloud function](https://console.cloud.google.com/functions/details/us-central1/serveGitHubRacTestFile?env=gen2&project=eppo-qa&tab=source). This makes sure the HTTPS methods work as expected (and is also required for Android emulated tests).